src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -2680,7 +2680,7 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
     // The following condition prevents using 16-byte stores when dstRegAddrAlignment is:
     //   1) unknown (i.e. dstReg is neither FP nor SP) or
     //   2) non-zero (i.e. dstRegAddr is not 16-byte aligned).
-    const bool hasAvailableSimdReg = isDstRegAddrAlignmentKnown && (size > FP_REGSIZE_BYTES);
```

Comment:
You will need to adjust LSRA. Otherwise, it won't allocate SIMD register(s) when src/dst doesn't use sp/fp as base register.

Reply:
Thanks @EgorChesakov . I believe that might be the reason SPC compilation is failing although, from what I understand, this was just the heuristics and we need to have to adjust LSRA.

Reply:
You need to adjust the following logic for InitBlock https://github.com/dotnet/runtime/blob/2d4f2d0c8f60d5f49e39f3ddbe1824648ee2b306/src/coreclr/jit/lsraarmarch.cpp#L632-L636
and for CopyBlock https://github.com/dotnet/runtime/blob/2d4f2d0c8f60d5f49e39f3ddbe1824648ee2b306/src/coreclr/jit/lsraarmarch.cpp#L711-L720